projects/storefrontapp-e2e-cypress/cypress/helpers/anonymous-consents.ts

```diff
@@ -0,0 +1,254 @@
+//   it('should restore anonoymous consents when logging out', () => {
+//     const newUser = registerUser();
+//     cy.visit(PAGE_URL_LOGIN);
+//     login(newUser.email, newUser.password);
+//     signOutUser();
+
+//     openDialogUsingFooterLink();
+//     giveOrRemoveAnonymousConsent(2);
+//     closeDialog();
+
+//     cy.visit(PAGE_URL_LOGIN);
+//     login(newUser.email, newUser.password);
+//     signOutUser();
+
+//     openDialogUsingFooterLink();
+//     cy.get('.cx-toggle-text:nth-child(2)').should('contain.text', 'ON');
+//   });
+// }
+
+export function changeLanguageTest() {
+  it('should pull the new consent templates but preserve the consents state', () => {
+    openDialogUsingFooterLink();
+    acceptAllConsent();
+
+    siteContextChange('/', LANGUAGES, LANGUAGE_DE, LANGUAGE_LABEL);
+    openDialogUsingFooterLink();
+    checkConsentsGiven();
+  });
+}
+
+export function registerUserAndCheckMyAccountConsent(
+  consentCheckBox,
+  position
+) {
+  const newUser = registerUser(consentCheckBox);
+  login(newUser.email, newUser.password);
+  checkBanner();
+  cy.selectUserMenuOption({
+    option: 'Consent Management',
+  });
+  cy.get(`input[type="checkbox"]:nth-child(${position})`)
+    .first()
+    .should('be.checked');
+}
+
+export function giveRegistrationConsentTest() {
+  it('should give the registration consent', () => {
+    registerUserAndCheckMyAccountConsent(true, 1);
+  });
+}
+
+export function testAsAnonymousUser() {
+  it('should be able to see the banner', () => {
+    seeBannerAsAnonymous();
+  });
+
+  it('should close the banner and give all consents by clicking on "ALLOW ALL" in the banner', () => {
+    clickAllowAllFromBanner();
+    checkBannerHidden();
+  });
+
+  it('should click the footer to check if all consents were accepted and withdraw all consents afterwards', () => {
+    // cy.wait(5000);
+    openDialogUsingFooterLink();
+
+    checkConsentsGiven();
+
+    rejectAllConsent();
+  });
+
+  it('should click the footer to check if all consents were rejected and accept all consents again', () => {
+    openDialogUsingFooterLink();
+
+    checkConsentsWithdrawn();
+
+    acceptAllConsent();
+  });
+
+  // I would put this here, but there's a problem (user can't login because it says dispatch had an error)
+  // it('should transfer anonoymous consents when registered', () => {
+  // openDialogUsingFooterLink();
+  // checkConsentsGiven();
+
+  // giveOrRemoveAnonymousConsent(2);
+  // giveOrRemoveAnonymousConsent(1);
+  // giveOrRemoveAnonymousConsent(3);
+  // closeDialog();
+
+  // registerUserAndCheckMyAccountConsent(false, 2);
+  // });
+
+  stub(LANGUAGE_REQUEST, LANGUAGES);
+
+  it('should pull the new consent templates but preserve the consents state', () => {
+    openDialogUsingFooterLink();
+    checkConsentsGiven();
+
+    siteContextChange('/', LANGUAGES, LANGUAGE_DE, LANGUAGE_LABEL);
+
+    openDialogUsingFooterLink();
+    checkConsentsGiven();
```

Comment:
The same question as in `changeLanguageTest()` - should we check for German consent templates?

Reply:
no need to check for language change because there's an e2e for that for consent in languages my account

Reply:
But that's for registered consents, this is for anonymous consent templates, no?